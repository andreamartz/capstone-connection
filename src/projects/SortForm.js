import React, { useState } from 'react';
import { useHistory } from "react-router-dom";
import { makeStyles } from '@material-ui/core/styles';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

const useStyles = makeStyles((theme) => ({
  button: {
    display: 'block',
    marginTop: theme.spacing(2),
  },
  formControl: {
    margin: theme.spacing(1),
    minWidth: 120,
  },
  select: {
    width: theme.spacing(20)
  }
}));

const SortForm = ({ sort, sortVariable, setSortVariable }) => {
  const [open, setOpen] = useState(false);
  const classes = useStyles();
  const history = useHistory();

  const handleChange = async (event) => {
    setSortVariable(sortVariable => event.target.value);
    await sort(sortVariable);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleOpen = () => {
    setOpen(true);
  };

  return (
    <FormControl className={classes.formControl}>
      <InputLabel id="sortVariable">Sort projects by</InputLabel>
      <Select
        labelId="sort variable"
        id="sortVariableSelect"
        open={open}
        onClose={handleClose}
        onOpen={handleOpen}
        value={sortVariable}
        onChange={handleChange}
        className={classes.select}
      >
        <MenuItem value="">
          <em>None</em>
        </MenuItem>
        <MenuItem 
          value={"newest"}
          onChange={handleChange}
        >
          Newest 
        </MenuItem>
        <MenuItem 
          value={"most likes"}
          onChange={handleChange}
        >
          Most Likes
        </MenuItem>
      </Select>
    </FormControl>
  );
}

export default SortForm; 