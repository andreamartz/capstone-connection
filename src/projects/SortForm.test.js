import React from 'react';
import { render } from '@testing-library/react';
import SortForm from './SortForm';

test('it renders without crashing', function () {
	render(<SortForm sortVariable={'most likes'} />);
});

test('it matches snapshot when sorting by most likes', function () {
	const { asFragment } = render(<SortForm sortVariable={'most likes'} />);
	expect(asFragment()).toMatchSnapshot();
});

test('it matches snapshot when sorting by most recently added projects', function () {
	const { asFragment } = render(<SortForm sortVariable={'newest'} />);
	expect(asFragment()).toMatchSnapshot();
});
