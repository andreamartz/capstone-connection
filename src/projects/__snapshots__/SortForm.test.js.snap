// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot when sorting by most likes 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root makeStyles-formControl-5"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      id="sortVariable"
    >
      Sort projects by
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-6 MuiInputBase-formControl MuiInput-formControl"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="sort variable sortVariableSelect"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="sortVariableSelect"
        role="button"
        tabindex="0"
      >
        Most Likes
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        tabindex="-1"
        value="most likes"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot when sorting by most recently added projects 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root makeStyles-formControl-8"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      id="sortVariable"
    >
      Sort projects by
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-9 MuiInputBase-formControl MuiInput-formControl"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="sort variable sortVariableSelect"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="sortVariableSelect"
        role="button"
        tabindex="0"
      >
        Newest
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        tabindex="-1"
        value="newest"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
